src/main/java/org/isf/admission/gui/PatientFolderReportModal.java

```diff
@@ -75,15 +75,36 @@
 	private JPanel allPanel;
 	private JCheckBox allCheck;
 	private JPanel labelPanel;
-	private GregorianCalendar date;
-	
-	public PatientFolderReportModal(JFrame parent, Integer code, GregorianCalendar olderDate) {
+	private GregorianCalendar fromDate;
+	private GregorianCalendar toDate;
+	private String selectedReport;
+
+	public PatientFolderReportModal(JFrame parent, Integer code, GregorianCalendar fromDate, GregorianCalendar toDate, String selectedReport) {
 		this.parent = parent;
 		this.patId = code;
-		this.date = olderDate;
+		this.fromDate = fromDate;
+		this.toDate = toDate;
+		this.selectedReport = selectedReport.toUpperCase();
+		switch(this.selectedReport) {
+			case "ADMISSION":
+			case "OPD":
+			case "LABORATORY":
+			case "OPERATION":
+			case "DRUGS":
+			case "EXAMINATION":
+				break;  // valid values
+			case "MALE WARD":
+			case "FEMALE WARD":
+			case "CHILDREN WARD":
+			case "MATERNITY WARD":
+				this.selectedReport = "ADMISSION";
+				break;
```

Comment:
there's no need of this because you can get the actual object from the selected row of each table (column = -1) and so use the instanceof to differenciate

Reply:
OK; I've implemented changes using column = -1, and fixed missing resource bundle